### ISSUE TITLE
Support SD Card Slot

### ISSUE THREAD

**Issue by seangineer** [utmapp/UTM#4243]
Add support for attaching an SD Card through the built-in card reader in some modern Macs. 

Currently, when plugging something into the slot, it is not detected by UTM, i.e. as a USB device to attach to the guest.

For reference, I am using a 2021 14" M1 Pro MacBook Pro.

**Comment by cicuz** [comment]
precisely what I'm trying to do... any workarounds in the meantime? :/

**Comment by wr1159** [comment]
any updates?

**Comment by asyba** [comment]
same I need to access sdcard

**Comment by guillaumekh** [comment]
QEMU supports this but I doubt Apple's virtualization framework supports it.
I cannot find anything that would enable such a feature in their docs — see the device types documented here: https://developer.apple.com/documentation/virtualization/vzvirtualmachineconfiguration#3667140
I'd love to be proven wrong though.

**Comment by asyba** [comment]
> QEMU supports this but I doubt Apple's virtualization framework supports it. I cannot find anything that would enable such a feature in their docs — see the device types documented here: https://developer.apple.com/documentation/virtualization/vzvirtualmachineconfiguration#3667140 I'd love to be proven wrong though.

yeah I was reading around the web and is something from Apple that prevents VM to read it.
I just bought a cheap SD-CARD reader to usb-c and it works on UTM I can read the sdcard

**Comment by wiegell** [comment]
My workaround for access to an ext4 sd card is to create an image of it:
```

# get the disk with 'diskutil list'
dd if=/dev/disk4 of=~/desktop/dump.img
# pwd = ~/desktop
qemu-img convert -f raw -O qcow2 dump.img dest.qcow2
```
then create a new disk for the machine in UTM and import the .qcow2 file.
I suppose [etcher](https://etcher.balena.io/) can write it back to a card, but i haven't tried that

**Comment by upintheairsheep** [comment]
QEMU supports many models of SD card slots, for now adding support for virtual sd cards would be nice.

**Comment by saschafoerster** [comment]
Just try to edit a Raspberry Pi SD-Card with UTM and Gparted Live and was stuck here.

**Comment by politician** [comment]
An external USB card reader will do the trick until this bug is fixed.
